### PR TITLE
JP Remote Install: Add tracks events

### PR DESCRIPTION
**Pull request by seear**
Add some analytics to the Jetpack Remote Install feature ahead of enabling in production. This should be enough events (combined with some existing JPC events) to track a site from first entry in the url box through to connection success (or not).

## Testing
* In your JS console, enter: `localStorage.debug = 'calypso:analytics:tracks'`
* Go to https://calypso.live/jetpack/connect?branch=add/remote-install-tracks
* Keep an eye on the console for the new tracks events as you proceed through the flow
* Enter the URL of a .org site without jetpack installed ( https://jurassic.ninja/create/?shortlived&nojetpack )
* Enter invalid credentials to cause an error response

### CODE REVIEW

File: client/state/data-layer/wpcom/jetpack-install/index.js [hunk 1]
```diff
@@ -47,8 +47,8 @@ export const handleError = ( action, error ) => {
 	const { retryCount = 0 } = dataLayer;
 
 	const logToTracks = withAnalytics(
-		recordTracksEvent( 'calypso_jpc_remote_install_api_response', {
-			remote_site_url: url,
+		recordTracksEvent( 'calypso_jpc_remote_install_api_failure', {
+			url: url,
```
Comment (tyxla) [inline client/state/data-layer/wpcom/jetpack-install/index.js]: We could shorten this to just `url`.

File: client/state/data-layer/wpcom/jetpack-install/index.js [hunk 2]
```diff
@@ -47,8 +47,8 @@ export const handleError = ( action, error ) => {
...
 			data: JSON.stringify( error ),
```
Comment (sirreal) [inline client/state/data-layer/wpcom/jetpack-install/index.js]: The error includes the request body in the error, let's make sure even bad credentials don't land in tracks:

```json
{
  "path": "/jetpack-install/…",
  "method": "POST",
  "apiVersion": "1.1",
  "query": "",
  "body": {
    "user": "demo",
    "password": "bad"
  },
  "callback": "tvh0y62",
  "supports_args": true,
  "supports_error_obj": true,
  "supports_progress": true,
  "name": "LoginFailureError",
  "statusCode": 400,
  "status": 400,
  "message": "",
  "error": "LOGIN_FAILURE"
}
```

Comment (seear) [reply client/state/data-layer/wpcom/jetpack-install/index.js]: Wow, nice spot :boom:

Let me sort that.

**Comment by seear** [comment]
Now following naming conventions, and only logging `error`, `message`, and `status` from API responses:

screen shot 2018-04-17 at 11 25 31

**Comment by seear** [comment]
Thanks for reviews everyone!
